src/garmin_parser.c

```diff
@@ -397,10 +398,19 @@ struct field_desc {
 	{ \
 		if (strcmp(#type, base_type_info[base_type].type_name)) \
 			fprintf(stderr, "%s: %s should be %s\n", #name, #type, base_type_info[base_type].type_name); \
-		type val = *(type *)p; \
+		type val; \
+		char is_endian_int = base_type_info[base_type].type_name[0] == 'S' || base_type_info[base_type].type_name[0] == 'U'; \
+		if (base_type_info[base_type].type_size > 1 && is_endian_int) { \
+			if (g->is_big_endian) \
+				val = array_uint_be(p ,base_type_info[base_type].type_size); \
+			else \
+				val = array_uint_le(p ,base_type_info[base_type].type_size); \
+		} else { \
+			val = *(type *)p; \
+		} \
```

Comment:
Can we add a "array_uint_endian(ptr, size, bigendian)" helper, and just use it here instead of this ugly thing?

Please?

Reply:
sure... there is lots of room for making this prettier.